packages/matchbox/src/components/TextField/TextField.js

```diff
@@ -1,179 +1,207 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { Label } from '../Label';
 import { Error } from '../Error';
 import { Connect } from '../Connect';
-import classnames from 'classnames';
-import styles from './TextField.module.scss';
-
-class TextField extends Component {
-  static displayName = 'TextField';
-
-  static propTypes = {
-    /**
-     * 'left' | 'center' | 'right'
-     */
-    align: PropTypes.oneOf(['left', 'center', 'right']),
-    id: PropTypes.string,
-    multiline: PropTypes.bool,
-    /**
-     * 'text' | 'email' | 'number' | 'password' | 'search' | 'tel' | 'url' | 'date'
-     */
-    type: PropTypes.oneOf(['text', 'email', 'number', 'password', 'search', 'tel', 'url', 'date']),
-    placeholder: PropTypes.string,
-    autoFocus: PropTypes.bool,
-    disabled: PropTypes.bool,
-    readOnly: PropTypes.bool,
-    required: PropTypes.bool,
-    /**
-     * 'none' | 'both' | 'horizontal' | 'vertical' | 'block' | 'inline'
-     */
-    resize: PropTypes.oneOf(['none', 'both', 'horizontal', 'vertical', 'block', 'inline']),
-    label: PropTypes.string,
-    labelHidden: PropTypes.bool,
-    helpText: PropTypes.node,
-    connectLeft: PropTypes.node,
-    connectRight: PropTypes.node,
-    prefix: PropTypes.node,
-    suffix: PropTypes.node,
-    error: PropTypes.string,
-    errorInLabel: PropTypes.bool,
-    onChange: PropTypes.func,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func
-  };
-
-  static defaultProps = {
-    resize: 'both',
-    type: 'text'
-  };
-
-  state = {
-    paddingLeft: null,
-    paddingRight: null
-  };
-
-  componentDidMount() {
-    this.calculatePadding();
-  }
-
-  componentDidUpdate({ prefix, suffix }) {
-    if (prefix !== this.props.prefix || suffix !== this.props.suffix) {
-      this.calculatePadding();
-    }
+import { Box } from '../Box';
+import { HelpText } from '../HelpText';
+import { getRectFor, roundToBaseline } from '../../helpers/geometry';
+import useInputDescribedBy from '../../hooks/useInputDescribedBy';
+import { tokens } from '@sparkpost/design-tokens';
+import { margin } from 'styled-system';
+import styled from 'styled-components';
+import { pick, omit } from '@styled-system/props';
+
+const StyledWrapper = styled('div')`
+  ${margin}
+`;
+
+// TODO Make this reusable and shared with Select
+const FieldBox = props => {
+  return (
+    <Box
+      as="input"
+      disabled={props.disabled}
+      width="100%"
+      border={props.hasError ? `1px solid ${tokens.color_red_700}` : '400'}
+      borderRadius="100"
+      bg={props.disabled ? 'gray.200' : 'white'}
```

Comment:
Maybe adding the `not-allowed` cursor would be a good idea when disabled?

Reply:
Added this one onto the `style` object to avoid pulling in styled-components for one rule